packages/block-library/src/image/index.php

```diff
@@ -252,10 +252,16 @@ function block_core_image_render_lightbox( $block_content, $block ) {
 	$q->set_attribute( 'data-wp-style--object-fit', 'selectors.core.image.lightboxObjectFit' );
 	$enlarged_image_content = $q->get_updated_html();
 
-	$background_color = esc_attr( wp_get_global_styles( array( 'color', 'background' ) ) );
+	// If the current theme does NOT have a `theme.json`, or the colors are not defined,
+	// we need to set the background color & close button color to some default values
+	// because we can't get them from the Global Styles.
+	$global_styles_color  = wp_get_global_styles( array( 'color' ) );
+	$has_background_color = ! empty( $global_styles_color['background'] );
+	$has_text_color       = ! empty( $global_styles_color['text'] );
+	$background_color     = esc_attr( ( wp_theme_has_theme_json() && $has_background_color ) ? $global_styles_color['background'] : '#fff' );
+	$close_button_color   = esc_attr( ( wp_theme_has_theme_json() && $has_text_color ) ? $global_styles_color['text'] : '#000' );
```

Comment:
Just call `wp_theme_has_theme_json()` once above to avoid duplicate calls

Reply:
If you don't have a `theme.json`, I think you shouldn't need to call `wp_get_global_styles()`. Furthermore, how about an approach like below to run the `wp_theme_has_theme_json()` function only once?

```php
$background_color   = '#fff';
$close_button_color = '#000';

if ( wp_theme_has_theme_json() ) {
	$global_styles_color = wp_get_global_styles( array( 'color' ) );
	if ( ! empty( $global_styles_color['background'] ) ) {
		$background_color = esc_attr( $global_styles_color['background'] );
	}
	if ( ! empty( $global_styles_color['text'] ) ) {
		$close_button_color = esc_attr( $global_styles_color['text'] );
	}
}
```